File: 백준/Silver/1439. 뒤집기/뒤집기.py
s = list(input())
cnt = 0
a = s[0]
for i in range(1,len(s)):
    if a != s[i]:
        cnt+=1
        a = s[i]

if cnt%2 == 1:
    if cnt>1:
        print(cnt-cnt//2)
    else : print(cnt)
else:
    print(cnt//2)
    

File: 백준/Silver/1439. 뒤집기/뒤집기.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;
        StringBuilder sb = new StringBuilder();
        String str = br.readLine();
        int cnt = 0; //바뀌는 부분 수
        char tmp = str.charAt(0);
        for (int i = 1; i < str.length(); i++) {
            if(tmp != str.charAt(i)){
                cnt++;
                tmp = str.charAt(i);
            }
        }
        if(cnt%2 == 1){
            cnt++;
        }
        cnt/=2;
        System.out.println(cnt);
    }
}